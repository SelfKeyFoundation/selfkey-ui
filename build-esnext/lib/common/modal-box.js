import * as React from 'react';
import { withStyles, createStyles } from '@material-ui/core';
import { Grid, Typography } from '@material-ui/core';
import { CloseButtonIcon } from '../icons/close-button';
const styles = createStyles({
    root: {
        boxSizing: 'border-box',
        minHeight: '300px',
        width: '781px',
        border: '1px solid #303C49',
        borderRadius: '4px',
        backgroundColor: '#262F39',
        boxShadow: 'inset -1px 0 0 0 rgba(0,0,0,0.24), 1px 0 0 0 rgba(118,128,147,0.2), 2px 0 2px 0 rgba(0,0,0,0.2)'
    },
    header: {
        height: '65px',
        width: '780px',
        borderRadius: '4px 4px 0 0',
        backgroundColor: '#2A3540',
        boxShadow: 'inset 0 -1px 0 0 #374758, 1px 0 0 0 rgba(118,128,147,0.2), 0 4px 2px 0 rgba(76,76,76,0.2)',
        '& h2': {
            height: '22px',
            width: '346px',
            color: '#FFFFFF',
            fontSize: '18px',
            lineHeight: '22px',
            marginLeft: '30px',
            marginTop: '20px'
        },
        overflow: 'visible',
        position: 'relative'
    },
    body: {
        padding: '30px 42px'
    },
    closeButton: {
        background: 'none',
        border: 'none',
        margin: 0,
        outline: 'none',
        padding: 0,
        position: 'absolute',
        top: '-18px',
        right: '-18px'
    }
});
export const ModalBox = withStyles(styles)((props) => (React.createElement(Grid, { container: true, className: props.classes.root, direction: "column", justify: "flex-start", alignItems: "flex-start" },
    React.createElement(Grid, { item: true, id: "header", className: props.classes.header },
        React.createElement(Typography, { variant: "h2" }, props.headerText),
        React.createElement("button", { className: props.classes.closeButton, onClick: props.closeAction },
            React.createElement(CloseButtonIcon, null))),
    React.createElement(Grid, { item: true, id: "body", className: props.classes.body }, props.children))));
export default ModalBox;
//# sourceMappingURL=modal-box.js.map