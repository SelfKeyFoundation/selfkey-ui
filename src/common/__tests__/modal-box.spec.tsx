import * as React from 'react';
import { create } from 'react-test-renderer';
import { ModalBox } from '../modal-box';

describe('Modal Box', () => {
  it('should show Modal Box', () => {
    const component = create(
      <ModalBox headerText='Modal boc'/>
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});





