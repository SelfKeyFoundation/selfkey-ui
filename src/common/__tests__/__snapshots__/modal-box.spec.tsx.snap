// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal Box should show Modal Box 1`] = `
<div
  className="MuiGrid-root Component-root-1 MuiGrid-container MuiGrid-direction-xs-column MuiGrid-align-items-xs-flex-start"
>
  <div
    className="MuiGrid-root Component-header-2 MuiGrid-item"
    id="header"
  >
    <h2
      className="MuiTypography-root MuiTypography-h2"
    >
      Modal boc
    </h2>
    <button
      className="Component-closeButton-4"
    >
      <svg
        aria-hidden="true"
        className="MuiSvgIcon-root Component-root-5"
        focusable="false"
        style={
          Object {
            "height": "38px",
            "width": "38px",
          }
        }
        viewBox="0 0 38 38"
      >
        <defs />
        <g
          fill="none"
          fillRule="evenodd"
          id="Symbols"
          stroke="none"
          strokeWidth="1"
        >
          <g
            id="Close-button"
            transform="translate(1.000000, 1.000000)"
          >
            <g
              id="Close"
            >
              <circle
                cx="18"
                cy="18"
                fill="#262F39"
                id="Oval-2"
                r="18"
              />
              <g
                fill="#23E6FE"
                fillRule="nonzero"
                id="icons8-delete_sign"
                transform="translate(9.000000, 9.000000)"
              >
                <polygon
                  id="Shape"
                  points="1.81534091 0.639204545 0.639204545 1.81534091 7.82386364 9 0.639204545 16.1846591 1.81534091 17.3607955 9 10.1761364 16.1846591 17.3607955 17.3607955 16.1846591 10.1761364 9 17.3607955 1.81534091 16.1846591 0.639204545 9 7.82386364"
                />
              </g>
            </g>
          </g>
        </g>
      </svg>
    </button>
  </div>
  <div
    className="MuiGrid-root Component-body-3 MuiGrid-item"
    id="body"
  />
</div>
`;
